.github/workflows/e2e-nightly-master.yml

```diff
@@ -34,11 +34,22 @@ jobs:
       - name: Generate testnets
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
-        run: ./build/generator -g 4 -d networks/nightly
+        run: ./build/generator -g 4 -d networks/nightly/legacy -p legacy
+        run: ./build/generator -g 4 -d networks/nightly/new -p new
+        run: ./build/generator -g 4 -d networks/nightly/split -p split
 
-      - name: Run testnets in group ${{ matrix.group }}
+      - name: Run legacy p2p testnets in group ${{ matrix.group }}
         working-directory: test/e2e
-        run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+        run: ./run-multiple.sh networks/nightly/legacy/*-group${{ matrix.group
+        }}-*.toml
+        
+      - name: Run new p2p testnets in group ${{ matrix.group }}
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks/nightly/new/*-group${{ matrix.group }}-*.toml
+        
+      - name: Run split p2p testnets in group ${{ matrix.group }}
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks/nightly/split/*-group${{ matrix.group }}-*.toml
```

Comment:
@marbar3778 can you check that this would work - I'm not so familiar with github actions

Reply:
this does look right, but you could simplify this with another matrix build of p2p stacks.

Reply:
if at line 19 you insert 
```
p2p: ['legacy', 'new', 'split']
```

you can replace where you specify these three to `${{ matrix.p2p }}`

this would reduce back to one line like it was before.

Reply:
ah right, thanks

Reply:
can I do this also with generating testnets i.e.
```
run: ./build/generator -g 4 -d networks/nightly/{{ matrix.p2p }} -p {{ matrix.p2p }} 
```

Reply:
yea that should work